clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -767,6 +767,14 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
     if (no_sanitize_mask & SanitizerKind::KernelHWAddress)
       SanOpts.set(SanitizerKind::HWAddress, false);
 
+    if (no_sanitize_mask && SanitizerKind::Realtime)
+    {
+      // If we have turned off the sanitizer for this method, what we really want is to bypass
+      // this function. Turn on the sanitizer, but add the attribute to bypass it.
+      SanOpts.set(SanitizerKind::Realtime, true);
+      Fn->addFnAttr(llvm::Attribute::RealtimeBypass);
+    }
```

Comment:
Alright, this feels like a kind of hack here.

Basically the idea is what I described in the original description of this PR. On line 1573, if the realtime sanitizer is disabled, it is turned off. That sounds stupid to type out, but it's important.

If the realtime sanitizer is turned off, the block on line 1573 is never tripped, and we never get the opportunity to add the bypass call in.


Our options:

1. Don't respect the __nosanitize attribute, which would mean we just remove this entire change from line 770-777, but remove some "expected behavior"
2. do this some other way?? Not sure how...


The psuedocode (implemented here) feels hacky:

```
if the sanititzer is disabled on this function:
     turn on the sanitizer  for this function
     add the "realtime bypass" attribute for this function
endif

```